Gems/ScriptCanvas/Code/Include/ScriptCanvas/Libraries/Spawning/SpawnableAsset.cpp

```diff
@@ -0,0 +1,134 @@
+                    ->Attribute(AZ::Edit::Attributes::AssetPickerTitle, "Spawnable Asset")
+                    ->Attribute(AZ::Edit::Attributes::ChangeNotify, &SpawnableAsset::OnSpawnAssetChanged);
+            }
+        }
+
+        if (auto behaviorContext = azrtti_cast<AZ::BehaviorContext*>(context))
+        {
+            behaviorContext
+                ->Class<SpawnableAsset>("SpawnableAsset")
+                ->Constructor()
+                ->Attribute(AZ::Script::Attributes::Scope, AZ::Script::Attributes::ScopeFlags::Common)
+                ->Attribute(AZ::Script::Attributes::Category, "Spawning")
+                ->Attribute(AZ::Script::Attributes::Module, "Spawning")
+                ->Property("m_asset", BehaviorValueProperty(&SpawnableAsset::m_asset));
+
+            behaviorContext
+                ->Class<ContainerTypeReflection::BehaviorClassReflection<SpawnableAsset>>(
+                    AZStd::string::format("ReflectOnDemandTargets_%s", Data::Traits<SpawnableAsset>::GetName().data()).data())
+                ->Attribute(AZ::Script::Attributes::ExcludeFrom, AZ::Script::Attributes::ExcludeFlags::All)
+                ->Attribute(AZ::Script::Attributes::Ignore, true)
+                // required to support Array<SpawnableAsset> variable type in Script Canvas
+                ->Method(
+                    "ReflectVector",
+                    [](const AZStd::vector<SpawnableAsset>&)
+                    {
+                    })
+                // required to support Map<String, SpawnableAsset> variable type in Script Canvas
+                ->Method(
+                    "Map_String_to_SpawnableAsset_Func",
+                    [](const AZStd::unordered_map<AZStd::string, SpawnableAsset>&)
+                    {
+                    })
+                // required to support Map<Number, SpawnableAsset> variable type in Script Canvas
+                ->Method(
+                    "Map_Number_to_SpawnableAsset_Func",
+                    [](const AZStd::unordered_map<double, SpawnableAsset>&)
+                    {
+                    });
+        }
+
+        ContainerTypeReflection::HashContainerReflector<SpawnableAsset>::Reflect(context);
+    }
+
+    SpawnableAsset::SpawnableAsset(const SpawnableAsset& rhs)
+        : m_asset(rhs.m_asset)
+    {
+    }
+
+    SpawnableAsset& SpawnableAsset::operator=(const SpawnableAsset& rhs)
+    {
+        m_asset = rhs.m_asset;
+        return *this;
+    }
+
+    void SpawnableAsset::OnSpawnAssetChanged()
+    {
+        if (m_asset.GetId().IsValid())
+        {
+            AZStd::string rootSpawnableFile;
+            AzFramework::StringFunc::Path::GetFileName(m_asset.GetHint().c_str(), rootSpawnableFile);
+
+            rootSpawnableFile += AzFramework::Spawnable::DotFileExtension;
+
+            AZ::u32 rootSubId = AzFramework::SpawnableAssetHandler::BuildSubId(move(rootSpawnableFile));
```

Comment:
Is this happening at runtime or is this just for use in the editor and/or builder.

Reply:
this is only when asset is assigned/changed, so no it's not happening in runtime.